src/Elastic.CommonSchema/Serialization/LogJsonFormatter .cs

```diff
@@ -3,13 +3,79 @@
 // See the LICENSE file in the project root for more information
 
 using System;
+using System.Collections.Generic;
 using Utf8Json;
 
 namespace Elastic.CommonSchema.Serialization
 {
-	internal class LogJsonFormatter : IJsonFormatter<Log>
+	internal partial class EcsUtf8JsonFormatterBase<TBase>
 	{
-		public Log Deserialize(ref JsonReader reader, IJsonFormatterResolver formatterResolver) => throw new NotImplementedException();
+		public bool ReadRef<T>(ref JsonReader reader, ref T set, IJsonFormatterResolver formatterResolver)
+		{
+			var formatter = formatterResolver.GetFormatter<T>();
+			set = formatter.Deserialize(ref reader, formatterResolver);
+			return true;
+		}
+		public bool Read<TProp>(ref JsonReader reader, TBase b, Action<TBase, TProp> set, IJsonFormatterResolver formatterResolver)
+		{
+			var formatter = formatterResolver.GetFormatter<TProp>();
+			set(b, formatter.Deserialize(ref reader, formatterResolver));
+			return true;
+		}
+
+		public static bool ReadString(ref JsonReader reader, ref string stringProp)
+		{
+			stringProp = reader.ReadString();
+			return true;
+		}
+	}
+
+
+	internal class LogJsonFormatter : EcsUtf8JsonFormatterBase<Log>, IJsonFormatter<Log>
+	{
+		private static IncrementingAutomataDictionary AutomataDictionary { get; } = new IncrementingAutomataDictionary
+		{
+			// Base fields
+			{ "origin" }, // 0
```

Comment:
Will follow up in a PR next week to generate this too... we dont want to miss fields.

Reply:
++